requirements.txt

```diff
@@ -0,0 +1,2 @@
+cssselect
+freezegun
```

Comment:
This is just a doubt, not a real issue:

If this is a test requirement, should it go to `test-requirements.txt`? Also, is it correct to have it in `external_dependencies` in the manifest if it is only used for the tests?

Reply:
> If this is a test requirement, should it go to `test-requirements.txt`?

Indeed, fixed.

> Also, is it correct to have it in `external_dependencies` in the manifest if it is only used for the tests?

I'm not sure about this. I couldn't find any guidelines in https://github.com/OCA/odoo-community.org/blob/master/website/Contribution/CONTRIBUTING.rst#external-dependencies and Odoo doesn't have a way to tell if a dependency is general or just for testing.

As for us, in doodba, it doesn't make a difference, because all test or non-test dependencies are installed anyways, I found it more useful to add it to the manifest, so CI tells if the module failure comes from a missing dependency. But I might be wrong... 🤷🏼‍♂️ @pedrobaeza?

Reply:
Well, I don't want to impose my criteria in this, but for me, it shouldn't be on external_dependencies, as it's only needed for tests, not for general install.

Reply:
Thanks for the insights :+1: 

@Yajo I see you moved the entire file to `test-requirements.txt`. Is `cssselect` only for the tests as well? I thought only `freezegun` was...

Reply:
In this case it would be a bit cumbersome to move them out of the manifest, because `cssselect` isn't imported into the tests. Instead, the features it adds implicitly to `lxml` are used. See https://github.com/lxml/lxml/blob/02a49b1d6ad177c948652f8b4d72aa0e2b386b89/CHANGES.txt#L1638-L1642

Also, if we remove the dependencies from the manifest, we wouldn't be able to specify valid version constraints, if/when any. Or, if we did, that info would be decontextualized from the module.

🤔 I think I'll just open one of those issues that Odoo will never fix, but just to keep track of this problem. But as long as we don't have a better place to put the dependencies, I think that putting them in the manifest will help more than harm.

Well, this is how it was before migrating anyway. Let's move on. 😄

Reply:
On v14, freezegun is included in Odoo core.